### DISCUSSION TITLE
Check for addin / tool updates

### DISCUSSION THREAD

**Discussion by GeertvanHorrik** [cake-build/cake#3055] [Q&A] [answered]
I am maintaining a [collection of shared build scripts](https://github.com/geertvanhorrik/repositorytemplate) where each file has its own addins / tools (so if we disable a file, it won't load / install the specific addins / tools required for this file).

However, it's a horrible task to check for updates of all the add-ins because I have to go through all the files, search in NuGet and update the version.

It would be fantastic if cake would have a "--versioncheck" (or something similar) that would return a json that:

1. Lists whether there is a new version of Cake itself
2. All available dependencies (and new versions if applicable)

Here is a quick mockup:

```
{
    "Cake": {
        "VersionInstalled": "1.0.0",
        "VersionAvailable": "1.0.1",
        "IsUpdateAvailable": true
    },
    "Addins": [
        {
            "Name": "Cake.Git",
            "VersionInstalled": "2.0.0",
            "VersionAvailable": "2.3.0",
            "IsUpdateAvailable": true
        },
        
        etc
    
    
    
    ],
    "Tools": 
    ... similar to addins

}
```

**Comment by nils-a** [comment, marked as answer]
@GeertvanHorrik Is that project hosted in GitHub and are the scripts filenames `*.cake`? If so https://github.com/nils-org/dependabot-cake-action could help you - it's a GitHub action to update references in Cake files, based on dependabot.

You can see the action in use here: https://github.com/cake-contrib/Cake.7zip/blob/develop/.github/workflows/dependabot-cake.yml
and the PR at https://github.com/cake-contrib/Cake.7zip/pull/92 was created using that Workflow

**Reply by GeertvanHorrik** [reply]
That is fantastic, thanks! Will try it out :)

**Reply by GeertvanHorrik** [reply]
It seems to work, thank you very much for your work and tips!

https://github.com/GeertvanHorrik/RepositoryTemplate/runs/1775589468?check_suite_focus=true

**Comment by augustoproiete** [comment]
@GeertvanHorrik your collection of Cake build scripts looks awesome! :star_struck:

Would you be interested in writing a little description of the scripts you have so far and how you're using them today, in a post in the [Show & Tell](https://github.com/cake-build/cake/discussions/categories/show-and-tell) here on GH discussions and/or publishing a blog post on the Cake website?

**Reply by GeertvanHorrik** [reply]
Sure, happy to do that. I will try to find some time later this week.
